### PR TITLE
ARC: cleanup instruction cache initialization

### PR DESCRIPTION
**Pull request by evgeniy-paltsev**
As of today during the Zephyr start we
 - invalidate I$
 - disable I$
 - enable I$

Given that we don't need to have I$ disabled during any initialization period and ARC processors have caches enabled after reset the I$ disabling/enabling is excessive, so we can drop it.

By that we also aligh the I$ initialization on ARC with other projects like U-boot and Linux kernel.

### CODE REVIEW

**Review by abrodkin — APPROVED** [review]
I'd reword the title as the major change here is removal of instruction cache.

**Review by abrodkin — APPROVED** [review]

**Comment by ruuddw** [comment]
Looks like an improvement, just not fully sure how it relates to Zephyr cache management (see also #30398)
E.g. in arc/arc/core/cache.c, I see: SYS_INIT(init_dcache, PRE_KERNEL_1, CONFIG_KERNEL_INIT_PRIORITY_DEFAULT), but nothing for icache.
Can you double check that cache init in reset.s and other kernel init is consistent? Instead of initializing in reset.s, it could be deferred to the C-level initialization (unless that is much later and has significant boot time impact). If init is handled in reset.S, I think the init in cache.c can go.

**Comment by evgeniy-paltsev** [comment]
> just not fully sure how it relates to Zephyr cache management (see also #30398)

We support D$ management in Zephyr for ARC. In #30398 it was converted to use standard Zephyr API for D$ management.

We do **not** support I$ management API in Zephyr for ARC. However we still do some I$ management in early ASM init code.
We do entire I$ invalidation to drop all previous code lines which may be cached. It may be useful in case of any pre-loader execution before Zephyr.

Before this PR we also did I$ disabling and re-enabling which is useless and can be avoided.

> E.g. in arc/arc/core/cache.c, I see: SYS_INIT(init_dcache, PRE_KERNEL_1, CONFIG_KERNEL_INIT_PRIORITY_DEFAULT), but nothing for icache.

Yep, in case of I$ we called initialization function (`z_icache_setup`) directly instead of init list usage.

> Can you double check that cache init in reset.s and other kernel init is consistent?

It's not actually (in the part related to D$ initialization). I'm planning to fix it in a separate pull request.

**Review by ruuddw — APPROVED** [review]
Ok (assuming consistent 'zephyr style' cache init done later in separate PR)
